examples/armbench-segmentation/leap_binder.py

```diff
@@ -336,6 +336,73 @@ def count_small_bbs(idx: int, data: PreprocessResponse) -> float:
     return float(len(areas[areas < CONFIG['SMALL_BBS_TH']]))
 
 
+def metadata_dict(idx: int, data: PreprocessResponse) -> Dict[str, Union[float, int, str]]:
+    metadata_functions = {
+        "idx": get_idx,
+        "fname": get_fname,
+        "origin_width": get_original_width,
+        "origin_height": get_original_height,
+        "instances_number": get_instances_num,
+        "tote_number": get_tote_instances_num,
+        "object_number": get_object_instances_num,
+        "avg_instance_size": get_avg_instance_percent,
+        "tote_instances_mean": get_tote_instances_mean,
+        "tote_instances_std": get_tote_instances_std,
+        "object_instances_mean": get_object_instances_mean,
+        "object_instances_std": get_object_instances_std,
+        "tote_avg_instance_size": get_tote_avg_instance_percent,
+        "tote_std_instance_size": get_tote_std_instance_percent,
+        "object_avg_instance_size": get_object_avg_instance_percent,
+        "object_std_instance_size": get_object_std_instance_percent,
+        "bbox_number": bbox_num,
+        "bbox_area": get_avg_bb_area,
+        "bbox_aspect_ratio": get_avg_bb_aspect_ratio,
+        "background_percent": get_background_percent,
+        "duplicate_bb": count_duplicate_bbs,
+        "small_bbs_number": count_small_bbs,
+        "count_total_obj_bbox_occlusions": get_obj_bbox_occlusions_count,
+        "avg_obj_bbox_occlusions": get_obj_bbox_occlusions_avg,
+        "count_obj_mask_occlusions": get_obj_mask_occlusions_count
+    }
+    res = dict()
+    for func_name, func in metadata_functions.items():
+        res[func_name] = func(idx, data)
+    return res
+
+
+def general_metrics_dict(bb_gt: tf.Tensor, detection_pred: tf.Tensor,
+                         mask_gt: tf.Tensor, segmentation_pred: tf.Tensor) -> Dict[str, tf.Tensor]:
+    metric_functions = {
+        "Regression_metric": regression_metric,
+        "Classification_metric": classification_metric,
+        "Objectness_metric": object_metric,
+        "Mask_metric": mask_metric,
+    }
+    res = dict()
+    for func_name, func in metric_functions.items():
+        res[func_name] = func(bb_gt, detection_pred, mask_gt, segmentation_pred)
+    return res
```

Comment:
This way, we are calculating many calculations four times; let's get into those functions and make sure we run efficiently.